lib/ansible/cli/galaxy.py

```diff
@@ -247,6 +253,14 @@ def init_parser(self):
         common.add_argument('-c', '--ignore-certs', action='store_true', dest='ignore_certs', help='Ignore SSL certificate validation errors.', default=None)
         common.add_argument('--timeout', dest='timeout', type=int,
                             help="The time to wait for operations against the galaxy server, defaults to 60s.")
+        common.add_argument('--collection-page-size', type=int, dest='collection_page_size',
+                            help='The page size to use when getting paginated collection results from a server. '
+                                 'Increasing the page size may decrease the number of API calls made when finding the best version to install/download.'
+                            )
+        common.add_argument('--role-page-size', type=int, dest='role_page_size',
+                            help='The page size to use when getting paginated role results from a server . '
```

Comment:
i would just have single 'page_size' since we already have roles/collections swtich in cli to determine which one we are looking at

Reply:
I added 2 separate ones since roles and collections can be installed together, wasn't sure if someone might want different page sizes for them. But I can condense if you think not.

Reply:
you can install both types on same run? https://github.com/ansible/ansible/issues/79442 would be solved then

Reply:
Yes, you can. I think #79442 is asking for a way to do that in a role (rather than managing the dependency manually in a requirements.yml).
 
You can install roles and collections together on the same run in a requirements file though and have the role use the collection. The dependency just needs to be managed manually.